test/unit/local/steps/dispatch.js

```diff
@@ -0,0 +1,389 @@
+  after('clean config directory', configHelpers.cleanConfig)
+
+  context('when buffer contains an initial-scan-done event', () => {
+    beforeEach(() => {
+      buffer.push([
+        builders.event().action('initial-scan-done').build()
+      ])
+    })
+
+    it('emits an initial-scan-done event via the emitter', async function () {
+      const dispatchedBuffer = dispatch(buffer, { events, prep, pouch: this.pouch })
+      await dispatchedBuffer.pop()
+
+      should(events.emit).have.been.calledOnce()
+      should(events.emit).have.been.calledWithExactly('initial-scan-done')
+    })
+
+    it('does not call any Prep method', async function () {
+      const dispatchedBuffer = dispatch(buffer, { events, prep, pouch: this.pouch })
+      await dispatchedBuffer.pop()
+
+      shouldHaveCalledOn(prep, [])
+    })
+  })
+
+  context('when buffer contains a scan file event', () => {
+    const filePath = 'foo'
+    const stats = { ino: 1, mtime: new Date(), ctime: new Date() }
+
+    beforeEach(() => {
+      buffer.push([
+        builders.event().action('scan').kind('file').path(filePath).stats(stats).build()
+      ])
+    })
+
+    it('triggers a call to addFileAsync with a file Metadata object', async function () {
+      const metafile = { path: filePath, ino: stats.ino, docType: 'file' }
+
+      const dispatchedBuffer = dispatch(buffer, { events, prep, pouch: this.pouch })
+      await dispatchedBuffer.pop()
+
+      shouldHaveCalledOn(prep, ['addFileAsync'])
+      should(prep.addFileAsync).have.been.calledOnce()
+      should(prep.addFileAsync).have.been.calledWithMatch('local', metafile)
+    })
+  })
+
+  context('when buffer contains a scan directory event', () => {
+    const directoryPath = 'foo'
+    const stats = { ino: 1, mtime: new Date(), ctime: new Date() }
+
+    beforeEach(() => {
+      buffer.push([
+        builders.event().action('scan').kind('directory').path(directoryPath).stats(stats).build()
+      ])
+    })
+
+    it('triggers a call to putFolderAsync with a directory Metadata object', async function () {
+      const metadir = { path: directoryPath, ino: stats.ino, docType: 'folder' }
+
+      const dispatchedBuffer = dispatch(buffer, { events, prep, pouch: this.pouch })
+      await dispatchedBuffer.pop()
+
+      shouldHaveCalledOn(prep, ['putFolderAsync'])
+      should(prep.putFolderAsync).have.been.calledOnce()
+      should(prep.putFolderAsync).have.been.calledWithMatch('local', metadir)
+    })
+  })
+
+  context('when buffer contains a created file event', () => {
+    const filePath = 'foo'
+    const stats = { ino: 1, mtime: new Date(), ctime: new Date() }
+
+    beforeEach(() => {
+      buffer.push([
+        builders.event().action('created').kind('file').path(filePath).stats(stats).build()
+      ])
+    })
+
+    it('triggers a call to addFileAsync with a file Metadata object', async function () {
+      const metafile = { path: filePath, ino: stats.ino, docType: 'file' }
```

Comment:
Maybe use `builders.metafile()`?
Maybe call it `doc` since it's the usual name in the codebase?

Reply:
I've tried to use `builders.metafile` at first to do an exact comparison with the actual parameter given to the `Prep` method but it was hard to get it right.
Since I moved to a partial comparison I thought it would be as convenient to create a partial `Metadata` object instead (it also shows it's not a complete match).

As for the name, I thought I'd have multiple documents in each test at first but since it's not the case I'll make the change.

Reply:
Couldn't [a helper](https://github.com/cozy-labs/cozy-desktop/pull/1407#discussion_r248702048) solve the comparison issue?